File: back/Ecopilot/src/main/java/com/ppp/Ecopilot/Entities/Transaction.java
package com.ppp.Ecopilot.Entities;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.validator.constraints.Length;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

@ToString
@SuperBuilder
@Entity
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class,
        property = "id")
public class Transaction extends BaseEntity<Long> {
    @Column(nullable = true, length = 40)
    private String description;
    @Column(nullable = true)
    private String address;
    @Column(nullable = false)
    private String country;
    @Column(nullable = false)
    @Min(value = 1000, message = "Le code postal doit être composé de 4 chiffres.")
    @Max(value = 9999, message = "Le code postal doit être composé de 4 chiffres.")
    private int postalCode = 1000;
    @Column(nullable = false)
    private String city;
    @Column(nullable = false)
    private String province;
    @Column(nullable = false)
    private float totalPrice;
    @Column(nullable = false,length = 8)
    private String phone;
    @NotEmpty(message = "chartLines is required")
    @OneToMany(mappedBy = "transaction",cascade = CascadeType.REMOVE)
    @JsonIgnore
    private List<ChartLine> chartLines;
    @ManyToOne()
    @JoinColumn(name = "companyOwner_id")
    @JsonIgnore
    private CompanyOwner companyOwner;
}

File: front/src/app/Models/transaction.ts
export class Transaction {
  public country: string;
  public flag: string;
  public quantity: number;
  public date: Date;
  public description: string;
  public chartLineIds: number[];
  public companyOwnerId: number;
  public totalPrice: number;
  public phone: string;
  public address: string;
  public postalCode: number;
  public city: string;
  public province: string;
  public createdAt: Date;

  constructor(country: string, flag: string, totalPrice: number, quantity: number, date: Date, description: string, chartLineIds: number[], companyOwnerId: number, phone: string, address: string, postalCode: number, city: string, province: string, createdAt: Date) {
    this.country = country;
    this.totalPrice = totalPrice;
    this.date = date;
    this.flag = flag;
    this.quantity = quantity;
    this.description = description;
    this.chartLineIds = chartLineIds;
    this.companyOwnerId = companyOwnerId;
    this.totalPrice = totalPrice;
    this.phone = phone;
    this.address = address;
    this.postalCode = postalCode;
    this.city = city;
    this.province = province;
    this.createdAt = createdAt;

  }
}

